From 475af64f64cbc210c87588d425fa9a14cd9760b6 Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Wed, 29 May 2019 11:38:53 +0300
Subject: [PATCH] Fix rpmfiles memory leak on %postuntrans file trigger
 preparation

---
 lib/rpmtriggers.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/rpmtriggers.c b/lib/rpmtriggers.c
index cb2441c5a..b7c76e7fc 100644
--- a/lib/rpmtriggers.c
+++ b/lib/rpmtriggers.c
@@ -154,6 +154,7 @@ void rpmtriggersPrepPostUnTransFileTrigs(rpmts ts, rpmte te)
 	}
     }
     rpmdbFreeIterator(mi);
+    rpmfilesFree(files);
 }
 
 int runPostUnTransFileTrigs(rpmts ts)
-- 
2.21.0

